### ISSUE TITLE
Query no longer returns results in InfluxDB 1.5.0

### ISSUE THREAD

**Issue by ot3** [influxdata/influxdb#9589]
I found another query that used to work pre 1.5.0. on exactly the same data
```
> SELECT DIFFERENCE(LAST("gas")) FROM "gas_time_shifted" WHERE time > 1475877600s AND time < 1475964000s GROUP BY time(1d) TZ('Europe/Amsterdam')
>
```
There are no results. To debug this previously working query I changed the GROUP BY time(1d) to GROUP BY time(1h) to prove that the data is present:
```
> SELECT DIFFERENCE(LAST("gas")) FROM "gas_time_shifted" WHERE time > 1475877600s AND time < 1475964000s GROUP BY time(1h) TZ('Europe/Amsterdam')
name: gas_time_shifted
time                      difference
----                      ----------
2016-10-08T02:00:00+02:00 0
2016-10-08T03:00:00+02:00 0
2016-10-08T04:00:00+02:00 0
2016-10-08T05:00:00+02:00 0
2016-10-08T06:00:00+02:00 0
2016-10-08T07:00:00+02:00 0
2016-10-08T08:00:00+02:00 0
2016-10-08T09:00:00+02:00 0.5349999999998545
2016-10-08T10:00:00+02:00 0.7860000000000582
2016-10-08T11:00:00+02:00 1.7090000000002874
2016-10-08T12:00:00+02:00 0.4849999999996726
2016-10-08T13:00:00+02:00 0
2016-10-08T14:00:00+02:00 0.11700000000018917
2016-10-08T15:00:00+02:00 0
2016-10-08T16:00:00+02:00 0.03099999999994907
2016-10-08T17:00:00+02:00 0
2016-10-08T18:00:00+02:00 0
2016-10-08T19:00:00+02:00 0.023000000000138243
2016-10-08T20:00:00+02:00 0.17699999999967986
2016-10-08T21:00:00+02:00 0.007000000000061846
2016-10-08T22:00:00+02:00 0
2016-10-08T23:00:00+02:00 0
>
```

I experimented with a slightly modified query where I added an "=" to the second time clause:

```
> SELECT DIFFERENCE(LAST("gas")) FROM "gas_time_shifted" WHERE time > 1475877600s AND time <= 1475964000s GROUP BY time(1d) TZ('Europe/Amsterdam')
name: gas_time_shifted
time                      difference
----                      ----------
2016-10-09T00:00:00+02:00 0
>
```
This time the result is 0 which is different from the pre 1.5.0. result of 3.870 which is the correct number for the total gas consumption for that day.

I have been unsuccessful finding a workaround; it would be really helpful to get one.

InfluxDB: 1.5.0
OS:  Ubuntu 17.10 (GNU/Linux 4.13.0-36-generic x86_64)

**Comment by dgnorton** [comment]
Possible duplicate of #9548

**Comment by dgnorton** [comment]
Fixed by #9566 for `1.5.1`

**Comment by ot3** [comment]
I installed 1.5.1. but found that the issue persists. I couldn't find any 1.5.1. release notes though, so don't know if it was supposed to be fixed.
